examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKerberizedKafkaWordCount.java

```diff
@@ -0,0 +1,138 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.streaming;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import scala.Tuple2;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.streaming.api.java.*;
+import org.apache.spark.streaming.kafka010.ConsumerStrategies;
+import org.apache.spark.streaming.kafka010.KafkaUtils;
+import org.apache.spark.streaming.kafka010.LocationStrategies;
+import org.apache.spark.streaming.Durations;
+
+/**
+ * Consumes messages from one or more topics in Kafka and does wordcount.
+ * Usage: JavaDirectKerberizedKafkaWordCount <brokers> <groupId> <topics>
+ *   <brokers> is a list of one or more Kafka brokers
+ *   <groupId> is a consumer group name to consume from topics
+ *   <topics> is a list of one or more kafka topics to consume from
+ *
+ * Example:
+ *   Yarn client:
+ *    $ bin/run-example --files ${jaas_path}/kafka_jaas.conf,${keytab_path}/kafka.service.keytab \
+ *      --driver-java-options "-Djava.security.auth.login.config=${path}/kafka_driver_jaas.conf" \
+ *      --conf \
+ *      "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./kafka_jaas.conf" \
+ *      --master yarn
+ *      streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
+ *      consumer-group topic1,topic2
+ *   Yarn cluster:
+ *    $ bin/run-example --files \
+ *      ${jaas_path}/kafka_jaas.conf,${keytab_path}/kafka.service.keytab,${krb5_path}/krb5.conf \
+ *      --driver-java-options \
+ *      "-Djava.security.auth.login.config=./kafka_jaas.conf \
+ *      -Djava.security.krb5.conf=./krb5.conf" \
+ *      --conf \
+ *      "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./kafka_jaas.conf" \
+ *      --master yarn --deploy-mode cluster \
+ *      streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
+ *      consumer-group topic1,topic2
+ *
+ * kafka_jaas.conf can manually create, template as:
+ *   KafkaClient {
+ *     com.sun.security.auth.module.Krb5LoginModule required
+ *     keyTab="./kafka.service.keytab"
+ *     useKeyTab=true
+ *     storeKey=true
+ *     useTicketCache=false
+ *     serviceName="kafka"
+ *     principal="kafka/host@EXAMPLE.COM";
+ *   };
+ * kafka_driver_jaas.conf (used by yarn client) and kafka_jaas.conf are basically the same
```

Comment:
If `--files` used for keytab and jaas file then both driver and executor can pick up the same jaas (in keytab file `./kafka.service.keytab` has to be set). Please see https://github.com/gaborgsomogyi/spark-structured-secure-kafka-app#spark-submit Is that not working somehow?

Reply:
> If `--files` used for keytab and jaas file then both driver and executor can pick up the same jaas (in keytab file `./kafka.service.keytab` has to be set). Please see https://github.com/gaborgsomogyi/spark-structured-secure-kafka-app#spark-submit Is that not working somehow?

I thought dirver can pick up the jaas, but it can't pick up files upload by `--files` after my test.
It can just work on cluster mode.